code/dataaccess/checkinputs.q

```diff
@@ -0,0 +1,194 @@
+  :checkparam/[dict;config];
+ };
+
+//- extract parameter specific function from confing - to check the input
+checkparam:{[dict;config] config[`checkfunction][dict;config`parameter]};
+
+//- generic function to takes in an atom/list of valid types and compare it against input types 
+checktype:{[validtypes;dict;parameter]
+  inputtype:type dict parameter;
+  if[not any validtypes~\:inputtype;'`$.dataaccess.formatstring["{parameter} input type incorrect - valid type(s):{validtypes} - input type:{inputtype}";`parameter`validtypes`inputtype!(parameter;validtypes;inputtype)]];
+  :dict;
+ };
+
+//- check if table exists + param is of type symbol
+//- if `tablename is in the correct format - we can then join on table properties from config
+isvalidtable:{[dict;parameter]
+  dict:issymbol[dict;parameter];
+  dict:tableexists[dict;parameter];
+  :.dataaccess.jointableproperties dict;
+ };
+
+tableexists:{[dict;parameter]
+  if[not dict[`tablename]in exec tablename from .dataaccess.tablepropertiesconfig;'`$.dataaccess.formatstring["table:{tablename} doesn't exist";dict]];
+  :dict;
+ };
+
+//- run checks to see if input columns exist
+checkcolumnsexist:{[dict;parameter]
+  dict:allsymbols[dict;parameter];
+  :columnsexist[dict;parameter;dict parameter];
+ };
+
+//- for a given list of columns check names against rdb/hdb metas
+columnsexist:{[dict;parameter;columns]
+  dict:checkinvalidcolumns[dict;parameter;columns;`rdbparams];
+  dict:checkinvalidcolumns[dict;parameter;columns;`hdbparams];
+  :dict;
+ };
+
+//- return error for any invalid columns
+checkinvalidcolumns:{[dict;parameter;columns;procparams]
+  if[not dict[procparams;`validrange];:dict];
+  validcolumns:exec columns from dict[procparams;`metainfo];
+  invalidcolumns:except[(),columns;validcolumns];
+  errorparams:dict,`parameter`proctype`validcolumns`invalidcolumns!(parameter;dict[procparams;`proctype];validcolumns;invalidcolumns);
+  if[count invalidcolumns;'`$.dataaccess.formatstring["parameter:{parameter} - table:{tablename} on process:{proctype} doesn't contain:{invalidcolumns} - validcolumns:{validcolumns}";errorparams]];
+  :dict;
+ };
+
+//- check starttime/endtime values are of valid type
+checktimetype:{[dict;parameter]:checktype[-12 -14 -15h;dict;parameter]};
+
+//- check param is of type symbol
+//- check starttime<=endtime
+//- use rollover function to split rdb/hdb range (rollover is the time data before which data is in the rdb)
+//- check the column exists for the given table
+//- check the given `timecolumn has valid type (e.g don't pass timecolumn:`sym)
+checktimecolumn:{[dict;parameter]
+  dict:issymbol[dict;parameter];
+  checktimeorder[dict;parameter];
+  dict:splitinputtime dict;
+  dict:columnsexist[dict;parameter;dict`timecolumn];
+  :checkcolumntype[dict;parameter;dict`timecolumn;-12 -14 -15h];
+ };
+
+checktimeorder:{[dict;parameter]
+  if[dict[`starttime]>dict`endtime;'`$"starttime>endtime"];
+  dict;
```

Comment:
Missing explicit return?

Reply:
Sorry Thomas - myself and Raveena are working off a branch called 'dataaccess.' I intended to merge in to that (I've closed this merge request to master).

Cheers for the heads up though.